### DISCUSSION TITLE
GraphQL: Accessible repositories for users (who are in a team)

### DISCUSSION THREAD

**Discussion by NiklasKK** [community/community#34447] [Apps, API and Webhooks]
I am trying to retrieve the repositories a user has access to using the GraphQL API. But it always returns an empty result.
But when I look at the user I am testing this with in the WebUI, I see that he has access to many repositories because he is a member of a team which where added to the respository.

**This is my GraphQL query:**
```
query {
                user(login: "username") {
                    repositories(first: 100, affiliations: ORGANIZATION_MEMBER) {
                        edges{
                            cursor
                            node{
                                name
                            }
                        }
                    }   
                }
          }
```

### The documentation for **ORGANIZATION_MEMBER** states:
```
"Repositories that the user has access to through being a member of an organization.
This includes every repository on every team that the user is on."
```

This is the result even though the WebUI confirms he has access to the repository due to his team membership.

`{"data":{"user":{"repositories":{"edges":[]}}}}`

Any ideas on this?

**Comment by LangLangBart** [comment]
`viewer` query with `affiliations`

> affiliations: [RepositoryAffiliation]
Array of viewer's affiliation options for repositories returned from the connection. 
For example, OWNER will include only repositories that the current **viewer** owns.

```zsh
{
  viewer {
    repositories(first: 100, affiliations: ORGANIZATION_MEMBER) {
      nodes {
        nameWithOwner
      }
    }
  }
}
```

`user` query with `ownerAffiliations`

> ownerAffiliations: [RepositoryAffiliation] 
Array of owner's affiliation options for repositories returned from the connection. 
For example, OWNER will include only repositories that the organization or **user** being viewed owns.

```zsh
{
  user(login: "ashtom") {
    repositories(first: 100, ownerAffiliations: ORGANIZATION_MEMBER) {
      nodes {
        nameWithOwner
      }
    }
  }
}
```

--- 

Why is it that way ?  
```zsh
¯\_(ツ)_/¯
```
